### PR TITLE
Add some ChefSpec tests

### PR DESCRIPTION
**Pull request by lanyonm**
This PR adds a couple ChefSpec tests.

As you can see, I had to include the `collectd::default` recipe in the `attribute_driven` recipe. This is because the `notifies :restart, "service[collectd]"` will error if used without the default recipe. Hopefully this change isn't objectionable.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'chefspec'
```
Comment (hectcastro) [inline spec/spec_helper.rb]: Would these dependencies need to be added to the `Gemfile` so that they can be installed to run the specs?

Comment (lanyonm) [inline spec/spec_helper.rb]: They certainly would. I've updated the Gemfile.ci so Travis will have these Gems installed.

Thanks for catching that.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+require 'chefspec/berkshelf'
+require 'support/matchers'
```
Comment (hectcastro) [inline spec/spec_helper.rb]: After installing the `chefspec` dependencies and running `rspec`, I was met with some errors until I commented out this line:

```
$ BUNDLE_GEMFILE=test/support/Gemfile.ci bundle exec rspec
/Users/hector/Projects/chef-collectd/spec/spec_helper.rb:3:in `require': cannot load such file -- support/matchers (LoadError)
	from /Users/hector/Projects/chef-collectd/spec/spec_helper.rb:3:in `<top (required)>'
	from /Users/hector/Projects/chef-collectd/spec/unit/attribute_driven_spec.rb:1:in `require'
	from /Users/hector/Projects/chef-collectd/spec/unit/attribute_driven_spec.rb:1:in `<top (required)>'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1226:in `load'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1226:in `block in load_spec_files'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1224:in `each'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1224:in `load_spec_files'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:97:in `setup'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:85:in `run'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:70:in `run'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:38:in `invoke'
	from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/exe/rspec:4:in `<top (required)>'
	from /usr/local/var/rbenv/versions/2.2.2/bin/rspec:23:in `load'
	from /usr/local/var/rbenv/versions/2.2.2/bin/rspec:23:in `<main>'
```

Comment (lanyonm) [inline spec/spec_helper.rb]: That require must have been from when I was experimenting on how to add it
- and I forgot to remove it. I've removed it and updated the tests with a
small enhancement that allows the ChefSpec Report to give a more accurate
measure of resource coverage.

Thanks,
Mike

On Sun, May 17, 2015 at 9:51 PM, Hector Castro <notifications@github.com>
wrote:

> In spec/spec_helper.rb
> <https://github.com/hectcastro/chef-collectd/pull/40#discussion_r30479271>
> :
>
> > @@ -0,0 +1,5 @@
> > +require 'chefspec'
> > +require 'chefspec/berkshelf'
> > +require 'support/matchers'
>
> After installing the chefspec dependencies and running rspec, I was met
> with some errors until I commented out this line:
>
> $ BUNDLE_GEMFILE=test/support/Gemfile.ci bundle exec rspec
> /Users/hector/Projects/chef-collectd/spec/spec_helper.rb:3:in `require': cannot load such file -- support/matchers (LoadError)
>     from /Users/hector/Projects/chef-collectd/spec/spec_helper.rb:3:in `<top (required)>'
>     from /Users/hector/Projects/chef-collectd/spec/unit/attribute_driven_spec.rb:1:in `require'
>     from /Users/hector/Projects/chef-collectd/spec/unit/attribute_driven_spec.rb:1:in `<top (required)>'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1226:in `load'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1226:in `block in load_spec_files'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1224:in `each'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/configuration.rb:1224:in `load_spec_files'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:97:in `setup'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:85:in `run'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:70:in `run'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/lib/rspec/core/runner.rb:38:in `invoke'
>     from /usr/local/var/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/rspec-core-3.2.3/exe/rspec:4:in `<top (required)>'
>     from /usr/local/var/rbenv/versions/2.2.2/bin/rspec:23:in `load'
>     from /usr/local/var/rbenv/versions/2.2.2/bin/rspec:23:in `<main>'
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hectcastro/chef-collectd/pull/40/files#r30479271>.
>



-- 
http://blog.lanyonm.org/
http://www.linkedin.com/in/lanyonm
https://github.com/lanyonm

**Comment by hectcastro** [comment]
:+1:

**Pull request merged**